src/Http/Routing/test/UnitTests/Builder/GroupTest.cs

```diff
@@ -301,7 +294,7 @@ public async Task ChangingMostEndpointBuilderPropertiesInConvention_Works()
 
         Assert.False(mapGetCalled);
         Assert.True(replacementCalled);
-        Assert.Equal("Replaced!", endpoint.DisplayName);
+        Assert.Equal("HTTP: GET Replaced!", endpoint.DisplayName);
```

Comment:
Is this a preferred change?

Reply:
Good eye! This is a natural consequence of group conventions now actually having a lower precedence than endpoint-specific conventions. In this case, we have the endpoint-specific convention being added by `MapMethods`.

https://github.com/dotnet/aspnetcore/blob/228ff77ba14d493d2765486f4f9726d50ad7dab3/src/Http/Routing/src/Builder/EndpointRouteBuilderExtensions.cs#L350-L351

So, I think this is the preferred behavior, and we just didn't notice that there was a group-specific convention that was getting clobbered before.

If we decided we wanted a group replacing the display name not to be wrapped by $"HTTP: {string.Join(", ", httpMethods)} {b.DisplayName}", it would actually be quite easy to do this. We would just move the wrapping logic straight into [the start of RouteEndpointDataSource.CreateRouteEndpointBuilder()](https://github.com/dotnet/aspnetcore/pull/42195/files#diff-b193e1f6a7930155befe00ee0d8d2e5c6e23951a1f5046c50e02d9c2527ac7e1R91-R100). The ease and flexibility of making that change is part of why I like adding the `RouteEndpointDataSource` so much.

Reply:
After further consideration, I decided ASP.NET developers aren't going to appreciate the nuance of the display name getting modified by an "endpoint-specific convention" vs just having that name to begin with. Since it was easy to fix, I did so. You can see the changes in [this commit](https://github.com/dotnet/aspnetcore/pull/42195/commits/1c5feef7f549e7bbf8c5c44a8724de30815864c0) which I think clean things up even further. EndpointRouteBuilderExtensions is now very simple!